adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/model/Datasource.java

```diff
@@ -11,123 +12,121 @@
+    return new AdapterConfig(
+      new ProtocolConfig(this.getProtocol().getType(), parameters),
+      new FormatConfig(this.getFormat().getType(), this.getFormat().getParameters())
+    );
   }
 
-    protected DatasourceProtocol fillQueryParameters(RuntimeParameters runtimeParameters) {
-      if (!this.getProtocol().getType().equals("HTTP")) {
-        return this.getProtocol();
-      }
-      String url = (String) this.getProtocol().getParameters().get("location");
-      Map<String, String> defaultParameters = new HashMap<>();
-      if (this.getProtocol().getParameters().containsKey("defaultParameters")) {
-        defaultParameters = (Map<String, String>) this.getProtocol().getParameters().get("defaultParameters");
-      }
-      Map<String, String> triggerParameters = new HashMap<>();
-      if (runtimeParameters != null && runtimeParameters.parameters != null) {
-        triggerParameters = runtimeParameters.parameters;
-      }
-      Map<String, String> replacementParameters = Stream.of(defaultParameters, triggerParameters).flatMap(map -> map.entrySet().stream()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (v1, v2) -> v2));
-      for (Map.Entry<String, String> parameter : replacementParameters.entrySet()) {
-        url = url.replace("{" + parameter.getKey() + "}", parameter.getValue());
-      }
-      HashMap<String, Object> parameters = new HashMap<>(this.getProtocol().getParameters());
-      parameters.put("location", url);
-      return new DatasourceProtocol(this.getProtocol().getType(), parameters);
+  protected Map<String, Object> fillQueryParameters(RuntimeParameters runtimeParameters) {
+    if (!this.getProtocol().getType().equals(Protocol.HTTP)) {
+      return this.getProtocol().getParameters();
     }
-
-    @Override
-    public boolean equals(Object o) {
-      if (this == o) return true;
-      if (!(o instanceof Datasource)) return false;
-      Datasource that = (Datasource) o;
-      return Objects.equals(id, that.id) &&
-        Objects.equals(protocol, that.protocol) &&
-        Objects.equals(format, that.format) &&
-        Objects.equals(metadata, that.metadata) &&
-        Objects.equals(trigger, that.trigger);
+    String url = (String) this.getProtocol().getParameters().get("location");
+    Map<String, String> defaultParameters = new HashMap<>();
+    if (this.getProtocol().getParameters().containsKey("defaultParameters")) {
+      defaultParameters = (Map<String, String>) this.getProtocol().getParameters().get("defaultParameters");
     }
+    Map<String, String> triggerParameters = new HashMap<>();
+    if (runtimeParameters != null && runtimeParameters.parameters != null) {
+      triggerParameters = runtimeParameters.parameters;
+    }
+    Map<String, String> replacementParameters = Stream.of(defaultParameters, triggerParameters).flatMap(map -> map.entrySet().stream()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (v1, v2) -> v2));
```

Comment:
This line is too long. You should always split stream API operation into separate lines.
```
Map<String, String> replacementParameters = Stream.of(defaultParameters, triggerParameters)
   .flatMap(map -> map.entrySet().stream())
   .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (v1, v2) -> v2));
```